### PR TITLE
KAFKA-8405; Remove deprecated `kafka-preferred-replica-election` command

### PR DESCRIPTION
**Pull request by dajac**
The `kafka-preferred-replica-election` command was deprecated in 2.4. As we removed the related APIs in the Admin client (https://github.com/apache/kafka/pull/10440), we can also remove the command line tool. `kafka-leader-election` can be used instead.

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Comment by dajac** [comment]
@ijuma What do you think about this one?

**Comment by ijuma** [comment]
Good question. I think this script is more widely used since it has existed for a lot longer than the admin commands we have removed in #10440. One question: could the script just point to the new command? Are the commands different enough that it would not work?

**Comment by hachikuji** [comment]
I'm somewhat more inclined to just get rid of it.

**Comment by chia7712** [comment]
+1 to get rid of it. Could you update `upgrade.html` to include the replacement of `kafka-preferred-replica-election`?

**Comment by dajac** [comment]
> Good question. I think this script is more widely used since it has existed for a lot longer than the admin commands we have removed in #10440. One question: could the script just point to the new command? Are the commands different enough that it would not work?

I just checked. It would still break current usages as the new script expect `--election-type` arg which is not in the old script. So existing users would have to adapt anyway. The semantic of `--path-to-json-file` is also slightly different in the two scripts. I think that this would be even more confusing for users.

**Comment by dajac** [comment]
I have updated the upgrade notes.

**Comment by ijuma** [comment]
Thinking about it, people would have to adjust to using the script with bootstrap-server and that was introduced much later, so I'm fine with removing.

File: docs/upgrade.html [hunk 1]
```diff
@@ -34,6 +34,7 @@ <h5><a id="upgrade_300_notable" href="#upgrade_300_notable">Notable changes in 3
     <li>The deprecated <code>Admin.electedPreferredLeaders()</code> methods were removed. Please use <code>Admin.electLeaders</code> instead.</li>
     <li>The deprecated <code>ConfigEntry</code> constructor was removed (<a href="https://issues.apache.org/jira/browse/KAFKA-12577">KAFKA-12577</a>).
         Please use the remaining public constructor instead.</li>
+    <li>The deprecated <code>kafka-preferred-replica-election</code> command line tool was removed. Please use <code>kafka-leader-election</code> instead.</li>
```
Comment (ijuma) [inline docs/upgrade.html]: I would move this just above or below the related admin API removal.

**Review by ijuma — APPROVED** [review]
LGTM, thanks. One nit below.

File: docs/upgrade.html (hunk 1 above)
Comment (dajac) [reply docs/upgrade.html]: Done.

**Comment by dajac** [comment]
Failed tests are not related. Merging it to trunk.

**Pull request merged**